### PR TITLE
[5.0] add option DateTime Class

### PR DESCRIPTION
**Pull request by thoscribe**
for Carbon or extension based on it such as jenssegers/laravel-date

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -133,6 +132,14 @@
 	protected $dates = array();
 
 	/**
+	 * Class for DateTime ( Carbon or extension based on it such as jenssegers/laravel-date ).
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: No spaces inside the brackets please.

File: src/Illuminate/Database/Eloquent/Model.php [hunk 2]
```diff
@@ -133,6 +132,14 @@
...
+	 *
+	 * @var string
+	 */
+
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: No extra line here please.

**Comment by neomerx** [comment]
Why overriding a property is better than overriding a method?

**Comment by GrahamCampbell** [comment]
Please squash to one commit.

**Comment by sycuato** [comment]
Done.

**Comment by JosephSilber** [comment]
I think it would be cleaner to just create a new `dateTime` method that returns a new instance of carbon, and use that.

That way if you want to change it you can override that method.

Just my $0.02.

**Comment by neomerx** [comment]
:-1: 
I don't like it because it adds one more object creation. Instead of "new Carbon/DateTime" it would be "new Carbon/DateTime -> new Carbon/DateTime"

**Comment by taylorotwell** [comment]
Sticking with Carbon.

**Pull request closed without merging**